src/nvim/textformat.c

```diff
@@ -56,7 +57,7 @@ static bool did_add_space = false;  ///< auto_format() added an extra space
 bool has_format_option(int x)
   FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  if (p_paste) {
+  if (p_paste || bt_prompt(curbuf)) {
```

Comment:
don't think we want special cases spread around. instead just unset formatoptions by defualt, which `buftype=prompt` is set. and mention this in this docs.

Reply:
I was thinking about that, but won't we loose user-set values for the option? 

Say someone does
```
set bt=prompt
...
set bt=
```

Doesn't that just unset the option? losing the older value? Is there a way to reset to last userset value?